### ISSUE TITLE
Resolve relative paths to files

### ISSUE THREAD

**Issue by mihaeu** [mihaeu/cosmere#12]
# Problem

[example](../another_file.md) does not resolve

**Comment by mihaeu** [comment]
Brainstorming:

```
if the relative link target is in the cosmere config
   generate a link to that confluence page
else if the relative link is part of the repository
  generate a link to the repository
else
  ???
``

**Comment by codethief** [comment]
Automatically converting links to relative paths would indeed be a very nice feature! I would give this a shot and try to come up with a patch, but so far I haven't been able to figure out where in the code links get converted. Could you give me a pointer?

> else if the relative link is part of the repository
>   generate a link to the repository

Which repository are you referring to?

**Comment by mihaeu** [comment]
Hi @codethief 

So I was assuming (based on how I use it and how I've seen others use it) that people keep their markdown files in a git repository.

So case 1 would be a lookup in the `cosmere.json` config. Case 2 would be a lookup in the repository that contains the file itself. An example might be that you're linking to a file in your codebase, e.g. `package.json` or `pom.xml` which will be lost in Confluence. But to really make these links useful, they would have to point to the repository. So maybe `<a href="https://github.com/mihaeu/cosmere/package.json">...</a>`, but the question is where to get the base URL from. It could be guessed from the git remote, but that's not reliable because if it's ssh then it's hard to extrapolate the URL from that. I think it would have to be another setting in the `cosmere.json`. Another option would be to lookup the property from the `package.json`. It's already in there, but I've also seen people use `cosmere` from a Docker container, globally or without really installing it using `npx` and then there wouldn't be a `package.json` to refer to.

Long story short: I guess a first step would be a simple config lookup. Step two could be adding a new field to the config and printing warnings if there are relative links that cannot be resolved.

And to make things even more complicated it would be nice if anchors would be resolved properly, because afaik that's not working properly right now.

**Comment by codethief** [comment]
Hmm I'm not sure I see the use case yet. What stops you from linking the Github URL directly in your Markdown files? I think what you're having in mind is a general way to re-write anchor links. That might definitely be useful but I think as a very first step I would focus on getting relative links (to other pages handled by Cosmere) right.

**Comment by mihaeu** [comment]
So what I meant was something like

```md
See [doc](doc.md)
```

If doc.md is also mirrored to confluence then I don't want them link to the
file I want the link to the confluence page.
